src/port1.0/portconfigure.tcl

```diff
@@ -515,17 +515,16 @@ proc portconfigure::configure_get_sdkroot {sdk_version} {
 
 # internal function to determine DEVELOPER_DIR according to Xcode dependency
 proc portconfigure::configure_get_developer_dir {} {
-    global use_xcode macosx_version
-    if {[vercmp $macosx_version 10.9] >= 0} {
-        set cltpath "/Library/Developer/CommandLineTools"
+    global use_xcode developer_dir
+    set cltpath "/Library/Developer/CommandLineTools"
+    # Assume that the existence of libxcselect indiciates the earliest version of
+    # macOS that places CLT in /Library/Developer/CommandLineTools
+    # If port is Xcode-dependent or CommandLineTools directory is invalid, set to developer_dir
+    if {[tbool use_xcode] || ![file exists /usr/lib/libxcselect.dylib] || ![file executable [file join $cltpath usr bin make]]} {
```

Comment:
Instead of `""`, we explicitly set `DEVELOPER_DIR` to `${developer_dir}` now if `use_xcode` or CLT is invalid. This covers the case that Xcode is installed but CLT isn't. As @jmroot pointed out, the check for `/usr/include` is unreliable so we check for only `/usr/bin/make`.